pkg/integrations/v2/oracledb_exporter/oracledb_exporter.go

```diff
@@ -0,0 +1,122 @@
+package oracledbexporter
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/agent/pkg/integrations/v2/common"
+	"github.com/grafana/agent/pkg/integrations/v2/metricsutils"
+	oe "github.com/observiq/oracledb_exporter/collector"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+
+	// required driver for integration
+	_ "github.com/sijms/go-ora/v2"
+	"github.com/xo/dburl"
+
+	integrations_v2 "github.com/grafana/agent/pkg/integrations/v2"
+)
+
+// DefaultConfig is the default config for the oracledb v2 integration
+var DefaultConfig = Config{
+	ConnectionString: os.Getenv("DATA_SOURCE_NAME"),
+	MaxOpenConns:     10,
+	MaxIdleConns:     0,
+	QueryTimeout:     5,
+	ScrapeInterval:   0,
+	Common:           common.MetricsConfig{},
+}
+
+var (
+	errNoConnectionString = errors.New("no connection string was provided")
+)
+
+// Config is the configuration for the oracledb v2 integration
+type Config struct {
+	ConnectionString  string               `yaml:"connection_string"`
+	MaxIdleConns      int                  `yaml:"max_idle_connections"`
+	MaxOpenConns      int                  `yaml:"max_open_connections"`
+	ScrapeInterval    time.Duration        `yaml:"scrape_interval"`
+	CustomMetricsPath string               `yaml:"custom_metrics_path"`
+	QueryTimeout      int                  `yaml:"query_timeout"`
+	Common            common.MetricsConfig `yaml:",inline"`
+}
+
+// UnmarshalYAML implements yaml.Unmarshaler for Config
+func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	*c = DefaultConfig
+	type plain Config
+	return unmarshal((*plain)(c))
+}
+
+// Validate returns errors if the configuration is invalid and the exporter could not function with it
+func (c *Config) Validate() error {
+	if c.ConnectionString == "" {
+		return errors.New("no connection string was provided")
+	}
+	if _, err := dburl.Parse(c.ConnectionString); err != nil {
+		return fmt.Errorf("unable to parse connection string: %w", err)
+	}
+	return nil
+}
```

Comment:
Is this called anywhere? I think it needs to be called manually in `NewIntegration` somewhere.

Reply:
Added 👍